packages/autocomplete-plugin-query-suggestions/src/createQuerySuggestionsPlugin.ts

```diff
@@ -125,12 +141,34 @@ export function createQuerySuggestionsPlugin<
                     > = [current];
 
                     if (i <= itemsWithCategories - 1) {
-                      const categories = getAttributeValueByPath(
-                        current,
-                        Array.isArray(categoryAttribute)
-                          ? categoryAttribute
-                          : [categoryAttribute]
-                      )
+                      let paths = (Array.isArray(categoryAttribute[0])
+                        ? categoryAttribute
+                        : [categoryAttribute]) as string[][];
+
+                      if (typeof categoryAttribute === 'string') {
+                        paths = [[categoryAttribute]];
+                      }
+
+                      const categoriesValues = paths.reduce<
+                        QuerySuggestionsFacetValue[]
+                      >((totalCategories, path) => {
+                        const attrVal = getAttributeValueByPath(current, path);
+
+                        if (!attrVal) {
+                          return totalCategories;
+                        }
+
+                        // use only the first facet value if multiple indexes needs to be targeted by multiple paths
+                        return totalCategories.concat(
+                          paths.length > 1 ? attrVal[0] : attrVal
+                        );
```

Comment:
Ok, I think it will be best to keep all the items — it will be just a generic way of accumulating categories, and the user will decide what should be done with it